markdown-mode.el

```diff
@@ -6099,10 +6106,23 @@ not at a link or the link reference is not defined returns nil."
     (match-string-no-properties 2))
    (t nil)))
 
-(defun markdown-follow-link-at-point ()
-  "Open the current non-wiki link in a browser."
-  (interactive)
-  (if (markdown-link-p) (browse-url (markdown-link-link))
+(defun markdown-link-filename (name)
+  "Remove fragment from file links."
+  (string-match markdown-regex-link-fragment name)
+  (match-string 1 name))
+
+(defun markdown-follow-link (name &optional other)
+  "Follow the link NAME in a browser.
+Open the with `find-file' when OTHER is non-nil."
+  (if other (find-file (markdown-link-filename name))
```

Comment:
On Wed, Apr 27, 2016 at 12:47:18AM -0700, Syohei YOSHIDA wrote:
> > +(defun markdown-follow-link (name &optional other)
> > +  "Follow the link NAME in a browser.
> > +Open the with `find-file' when OTHER is non-nil."
> > +  (if other (find-file (markdown-link-filename name))
>
> Is `ffap` better rather than `find-file` ? If the link is HTTP
> URL(like `http://github.com/docker/docker/blob/master/README.md`),
> `C-u M-x markdown-follow-link-at-point` creates terrible file.

Ah, that sounds like the “find-file vs. browse-url logic” I wished for
in the initial comment [1].  However, in this case the target is a
name, not at the point.  Ideally we'd have a
find-file-with-ffap-url-regexp.  Does a function like that exist?
Should we use ffap-url-regexp ourselves?

With the current patch, I'm leaving it up to the user to C-u as
appropriate, but with reference-style links, they may not have the
link-target available to inform their choice.

[1]: https://github.com/jrblevin/markdown-mode/pull/132#issue-151304356